src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java

```diff
@@ -722,12 +723,15 @@ public void onPlayerCommandPreprocess(PlayerCommandPreprocessEvent event)
             // CommandBlocker handles messages and broadcasts
             event.setCancelled(true);
         }
-
+        
+        // Grab adminlist values
+        TFM_Admin admin = TFM_AdminList.getEntry(player);
+        
         if (!TFM_AdminList.isSuperAdmin(player))
         {
             for (Player pl : Bukkit.getOnlinePlayers())
             {
-                if (TFM_AdminList.isSuperAdmin(pl) && TFM_PlayerData.getPlayerData(pl).cmdspyEnabled())
+                if (TFM_AdminList.isSuperAdmin(pl) && admin.cmdSpyEnabled())
```

Comment:
@tylerhyperHD Follow the logic:
* `admin` refers to the entry for the player executing the command, not the looped player.
* `admin` may be null.